tests/models/test_instrument.py

```diff
@@ -290,5 +296,102 @@ def test_demoim_search_notimplemented():
+            assert info[ 'fringe_set' ] == 'DECam Default'
+        with SmartSession() as session:
+            im = session.get( Image, info[ 'flat_fileid' ] )
+            assert im.filter == filt
+            # The linearity file is the same for all sensor sections in DECam
+            if linfile is None:
+                linfile = session.get( DataFile, info[ 'linearity_fileid' ] )
+            else:
+                linfile = linfile.recursive_merge( session )
+                assert linfile == session.get( DataFile, info[ 'linearity_fileid' ] )
+
+
+def test_overscan_sections( decam_example_raw_image ):
+    decam = get_instrument_instance( "DECam" )
+    # Need the full header, not what's stored in the database
+    ovsecs = decam.overscan_sections( decam_example_raw_image.raw_header )
+    assert ovsecs == [ { 'secname': 'A',
+                         'biassec' : { 'x0': 6, 'x1': 56, 'y0': 0, 'y1': 4096 },
+                         'datasec' : { 'x0': 56, 'x1': 1080, 'y0': 0, 'y1': 4096 }
+                        },
+                       { 'secname': 'B',
+                         'biassec': { 'x0': 2104, 'x1': 2154, 'y0': 0, 'y1': 4096 },
+                         'datasec': { 'x0': 1080, 'x1': 2104, 'y0': 0, 'y1': 4096 }
+                        } ]
+
+def test_overscan_and_data_sections( decam_example_raw_image ):
+    decam = get_instrument_instance( "DECam" )
+    # Need the full header, not what's stored in the database
+    ovsecs = decam.overscan_and_data_sections( decam_example_raw_image.raw_header )
+    assert ovsecs == [ { 'secname': 'A',
+                         'biassec' : { 'x0': 6, 'x1': 56, 'y0': 0, 'y1': 4096 },
+                         'datasec' : { 'x0': 56, 'x1': 1080, 'y0': 0, 'y1': 4096 },
+                         'destsec' : { 'x0': 0, 'x1': 1024, 'y0': 0, 'y1': 4096 }
+                        },
+                       { 'secname': 'B',
+                         'biassec': { 'x0': 2104, 'x1': 2154, 'y0': 0, 'y1': 4096 },
+                         'datasec': { 'x0': 1080, 'x1': 2104, 'y0': 0, 'y1': 4096 },
+                         'destsec': { 'x0': 1024, 'x1': 2048, 'y0': 0, 'y1': 4096 }
+                        } ]
+
+def test_overscan( decam_example_raw_image ):
+    decam = get_instrument_instance( "DECam" )
+
+    # Make sure it fails if it gets bad arguments
+    with pytest.raises( TypeError, match='overscan_and_trim: pass either an Image as one argument' ):
+        junk = decam.overscan_and_trim( 42 )
+    with pytest.raises( RuntimeError, match='overscan_and_trim: pass either an Image as one argument' ):
+        junk = decam.overscan_and_trim()
+    with pytest.raises( RuntimeError, match='overscan_and_trim: pass either an Image as one argument' ):
+        junk = decam.overscan_and_trim( 1, 2, 3 )
+    with pytest.raises( TypeError, match="data isn't a numpy array" ):
+        junk = decam.overscan_and_trim( decam_example_raw_image.raw_header, 42 )
```

Comment:
I think the convention is to assign into `_`

Reply:
Back in the day (late 1990s), the language IDL would not *let* you call a function with return parameters without setting that return parmeter to something.  So, you *had* to store the result somewhere, even if you didn't care.  I wrote in a whole bunch of "shit = function_call(...)" into the code.  At some point, Michael Wood-Vasey went through and replaced all of them with "stuff = function_call(...)"